libmamba/src/core/package_handling.cpp

```diff
@@ -354,14 +356,53 @@ namespace mamba
         LOG_INFO << "Extracting " << file << " to " << destination;
         extraction_guard g(destination);
 
+        struct archive* a;
+        int r;
+
+        a = archive_read_new();
+        archive_read_support_format_tar(a);
+        archive_read_support_format_zip(a);
+        archive_read_support_filter_all(a);
+
+        auto lock = LockFile(file);
+        r = archive_read_open_filename(a, file.string().c_str(), 10240);
+
+        if (r != ARCHIVE_OK)
+        {
+            LOG_ERROR << "Error opening archive: " << archive_error_string(a);
+            throw std::runtime_error(file.string() + " : Could not open archive for reading.");
+        }
+
+        stream_extract_archive(a, destination);
+
+        archive_read_close(a);
+        archive_read_free(a);
```

Comment:
This definitely need to be RAII-ed/scoped as there is a potential exception just before.
Consider using  the `mamba::on_scope_exit()` for this.